internal/util/search/search.go

```diff
@@ -204,83 +202,47 @@ func (sr *SearchReplace) putLiteral(object *yaml.RNode) error {
 	res := SearchResult{
 		FilePath:  sr.filePath,
 		FieldPath: sr.ByPath,
-		Value:     sr.PutLiteral,
+		Value:     sr.PutValue,
 	}
 	sr.Result = append(sr.Result, res)
 	sr.Count++
 	return nil
 }
 
-// shouldPutLiteralByPath returns true if only absolute path and literal are provided,
+// shouldPutValueByPath returns true if only absolute path and literal are provided,
 // so that the value can be directly put without needing to traverse the entire node,
 // handles the case of adding non-existent field-value to node
-func (sr *SearchReplace) shouldPutLiteralByPath() bool {
+func (sr *SearchReplace) shouldPutValueByPath() bool {
 	return isAbsPath(sr.ByPath) &&
 		!strings.Contains(sr.ByPath, "[") && // TODO: pmarupaka Support appending literal for arrays
 		sr.ByValue == "" &&
 		sr.ByValueRegex == "" &&
-		sr.PutLiteral != ""
+		sr.PutValue != ""
 }
 
-// setterPatternFromRegex takes the field value of a node, valueRegex provided by
-// user from --by-value-regex, patternRegex provided by user from --put-pattern,
-// and makes best effort to derive the corresponding setter pattern comment to be
-// added as line comment to the node.
-// e.g. fieldValue: my-project-foo, valueRegex: my-project-*, patternRegex: ${project-id}-*
-// the output pattern will be ${project-id}-foo
-// in case the valueRegex is vague and not enough to derive the setter values, it
-// returns an error
-func setterPatternFromRegex(fieldValue, valueRegex, patternRegex string) (string, error) {
-	settersValues, err := settersValues(valueRegex, patternRegex)
-	if err != nil {
-		return "", err
+// resolvePattern takes the field value of a node, valueRegex provided by
+// user from --by-value-regex, patternRegex provided by user from --put-value/--put-comment,
+// and makes best effort to derive the corresponding capture groups and resolve the pattern
+// refer to tests for expected behavior
+func resolvePattern(fieldValue, valueRegex, patternRegex string) (string, error) {
+	if valueRegex == "" {
+		return patternRegex, nil
 	}
-
-	pattern := fieldValue
-	// derive the pattern from the field value by replacing setter values
-	// with setter name markers
-	// e.g. if field value is "my-project-foo", input PutPattern is "${project]-*", and
-	// value for setter "project" is "my-project",
-	// the pattern to be added as comment is ${project]-foo
-	for sn, sv := range settersValues {
-		pattern = strings.ReplaceAll(clean(pattern), clean(sv), clean(sn))
+	r := regexp.MustCompile(valueRegex)
```

Comment:
does this panic if it doesn't compile?

Reply:
Yes. Updated PR to properly wrap and throw error and added test.